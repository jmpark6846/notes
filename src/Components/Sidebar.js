import React from "react";
import { Pane, Menu, Text, Position, Popover, Avatar } from "evergreen-ui";
import { userContext } from "../Context";


const Sidebar = props => {
  return (
    <userContext.Consumer>
      {user => (
        <Pane height="100%" background="tint1" data-testid="sidebar" id="sidebar">
          {/* display: flex */}
          <Menu>
            <Pane>
              <Menu.Group>
                <Popover
                  position={Position.BOTTOM_RIGHT}
                  content={
                    <Menu>
                      <Menu.Group>
                        <Menu.Item
                          icon="log-out"
                          onSelect={props.onLogout}
                        >
                          로그아웃
                        </Menu.Item>
                      </Menu.Group>
                    </Menu>
                  }
                >
                  <Menu.Item>
                    <Pane display="flex" alignItems="center">
                      <Avatar
                        name={user.username}
                        size={25}
                        sizeLimitOneCharacter={25}
                        marginRight={5}
                      />
                      <Text fontWeight={600}>{user.username}</Text>
                    </Pane>
                  </Menu.Item>
                </Popover>
              </Menu.Group>
            </Pane>
            <Pane
              className="note-list"
              flex={1}
              overflowX="hidden"
              overflowY="auto"
            >
              <Menu.Group>
                {props.noteList.map(note => (
                  <Menu.Item
                    key={note.id}
                    onSelect={() => props.onNoteSelect(note.id)}
                  >
                    <Text
                      fontWeight={note.id === props.selected ? 700 : 500}
                    >
                      {note.title || "제목 없음"}
                    </Text>
                  </Menu.Item>
                ))}
              </Menu.Group>
            </Pane>
            <Pane>
              <Menu.Group>
                <Menu.Item onSelect={props.onAddNote} icon="edit">
                  새 노트 작성하기
                </Menu.Item>
              </Menu.Group>
            </Pane>
          </Menu>
        </Pane>
      )}
    </userContext.Consumer>
  );
};

export default Sidebar;
