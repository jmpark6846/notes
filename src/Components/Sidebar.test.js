import React from "react";
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Sidebar from "./Sidebar";

afterEach(cleanup)

test('matches snapshot ', () => {
  const { asFragment } = render(<Sidebar noteList={[]}/>)
  expect(asFragment()).toMatchSnapshot()
})

