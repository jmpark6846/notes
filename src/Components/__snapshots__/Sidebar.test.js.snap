// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot  1`] = `
<DocumentFragment>
  <div
    class="ðŸ“¦bg_1vd85pq ðŸ“¦h_100prcnt ðŸ“¦box-szg_border-box css-nil"
    data-testid="sidebar"
    id="sidebar"
  >
    <nav
      class="ðŸ“¦box-szg_border-box css-nil"
      outline="none"
      role="menu"
    >
      <div
        class="ðŸ“¦box-szg_border-box css-nil"
      >
        <div
          class="ðŸ“¦pb_8px ðŸ“¦pt_8px ðŸ“¦box-szg_border-box css-nil"
        >
          <div
            aria-haspopup="true"
            class="css-z3ro6d ðŸ“¦h_32px ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦box-szg_border-box css-nil"
            data-isselectable="true"
            role="button"
            tabindex="0"
          >
            <span
              class="ðŸ“¦color_none ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦ml_16px ðŸ“¦mr_16px ðŸ“¦flx_1 ðŸ“¦box-szg_border-box"
            >
              <div
                class="ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦box-szg_border-box css-nil"
              >
                <div
                  class="ðŸ“¦w_25px ðŸ“¦h_25px ðŸ“¦ovflw-x_hidden ðŸ“¦ovflw-y_hidden ðŸ“¦bblr_9999px ðŸ“¦bbrr_9999px ðŸ“¦btlr_9999px ðŸ“¦btrr_9999px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦flx-srnk_0 ðŸ“¦just-cnt_center ðŸ“¦bg-clr_eae7f8 ðŸ“¦mr_5px ðŸ“¦box-szg_border-box"
                >
                  <span
                    class="ðŸ“¦color_37248f ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_12px ðŸ“¦ltr-spc_-0-05px ðŸ“¦w_25px ðŸ“¦h_25px ðŸ“¦box-szg_border-box css-1ktdej1"
                  >
                    ?
                  </span>
                </div>
                <span
                  class="ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_600 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦box-szg_border-box"
                />
              </div>
            </span>
          </div>
        </div>
      </div>
      <div
        class="note-list ðŸ“¦flx_1 ðŸ“¦ovflw-x_hidden ðŸ“¦ovflw-y_auto ðŸ“¦box-szg_border-box css-nil"
      >
        <div
          class="ðŸ“¦pb_8px ðŸ“¦pt_8px ðŸ“¦box-szg_border-box css-nil"
        />
      </div>
      <div
        class="ðŸ“¦box-szg_border-box css-nil"
      >
        <div
          class="ðŸ“¦pb_8px ðŸ“¦pt_8px ðŸ“¦box-szg_border-box css-nil"
        >
          <div
            class="css-z3ro6d ðŸ“¦h_40px ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦box-szg_border-box css-nil"
            data-isselectable="true"
            role="menuitem"
            tabindex="0"
          >
            <svg
              class="ðŸ“¦ml_16px ðŸ“¦mr_-4px ðŸ“¦flx-srnk_0 ðŸ“¦w_16px ðŸ“¦h_16px ðŸ“¦box-szg_border-box"
              data-icon="edit"
              style="fill: #66788A;"
              viewBox="0 0 16 16"
            >
              <path
                d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                fill-rule="evenodd"
              />
            </svg>
            <span
              class="ðŸ“¦color_none ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_14px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_20px ðŸ“¦ltr-spc_-0-05px ðŸ“¦ml_16px ðŸ“¦mr_16px ðŸ“¦flx_1 ðŸ“¦box-szg_border-box"
            >
              ìƒˆ ë…¸íŠ¸ ìž‘ì„±í•˜ê¸°
            </span>
          </div>
        </div>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
